### PR TITLE
ui

### PR DESCRIPTION
**Pull request by rajdeepbharati**
Closes https://github.com/macports/buildbot-macports-custom-views/issues/4, https://github.com/macports/buildbot-macports-custom-views/issues/3

@tardyp

Screenshot 2019-06-23 at 11 24 01 AM
Screenshot 2019-06-23 at 11 23 00 AM
Screenshot 2019-06-23 at 11 25 44 AM

### CODE REVIEW

**Comment by mojca** [comment]
It could make sense to attach a screenshot or two in this PR (more or less the same as what you sent via email, but of course the one for the latest version).

File: src/components/Change.vue [hunk 1]
```diff
@@ -1,45 +1,45 @@
 <template>
   <div class="change-modal">
     <div class="modal-header">
-      <button
+      <!-- <button
```
Comment (p12tic) [inline src/components/Change.vue]: Worth just to delete no longer used code? You can go back in history with git if you ever need it.

Comment (rajdeepbharati) [reply src/components/Change.vue]: Done

File: src/components/FilterBuilds.vue [hunk 2]
```diff
@@ -124,11 +194,62 @@ export default {
       this.$data.rdate = date.getTime() / 1000
       console.log('final ' + this.$data.rdate)
     },
-    dt: function(timestamp) {
-      return new Date(timestamp * 1000)
+    formatDate: function(timestamp) {
+      var d = new Date(timestamp * 1000)
```
Comment (tardyp) [inline src/components/FilterBuilds.vue]: moment is supposed to help for this kind of code.
You should be able to use it by requiring MOMENT in the angularjs dependency injection of your directive.
You can also use DurationFormat and DateFormat for a standard way of formating dates in buildbot

https://github.com/buildbot/buildbot/blob/d913f1ec10dc7b751d414c090b83644c192d85a4/www/base/src/app/common/filters/moment/moment.filter.js#L13

They are also available as dependency injection.

File: src/components/PortsList.vue [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        <span class="msg">Crunching latest data...</span>
+      </div>
+      <div v-else>
+        <input type="text" v-model="search" />
+        <br />
+        <div>
+          <Port :portname="location.search()['id']" />
+        </div>
+        <div
+          class="btn-group-verticalq"
```
Comment (tardyp) [inline src/components/PortsList.vue]: typo?

File: src/ports.directive.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        sourcestamps: []
+      }
+
+      var e = new ComponentClass({
+        data: data,
+        el: element.get(0)
+      })
+
+      function update() {
+        data.location = $location
```
Comment (tardyp) [inline src/ports.directive.js]: At some point of the project, which should find something a bit more efficient.
This is not scalable to copy the whole data each time one of those collection change.
This is not needed now, but please keep in mind

File: src/components/PortsList.vue (hunk 3 above)
Comment (rajdeepbharati) [reply src/components/PortsList.vue]: Some css needs to be added in this.

File: src/ports.directive.js (hunk 4 above)
Comment (rajdeepbharati) [reply src/ports.directive.js]: Okay, noted.

File: src/components/FilterBuilds.vue (hunk 2 above)
Comment (rajdeepbharati) [reply src/components/FilterBuilds.vue]: Okay, I will change that in the next pr

**Pull request merged**